src/NuGet.Core/NuGet.CommandLine.XPlat/Native/PlatformApis.cs

```diff
@@ -0,0 +1,221 @@
+        {
+            Version version;
+            var kernelRelease = NativeMethods.Darwin.GetKernelRelease();
+            if (!Version.TryParse(kernelRelease, out version) || version.Major < 5)
+            {
+                // 10.0 covers all versions prior to Darwin 5
+                // Similarly, if the version is not a valid version number, but we have still detected that it is Darwin, we just assume
+                // it is OS X 10.0
+                return "10.0";
+            }
+            else
+            {
+                // Mac OS X 10.1 mapped to Darwin 5.x, and the mapping continues that way
+                // So just subtract 4 from the Darwin version.
+                // https://en.wikipedia.org/wiki/Darwin_%28operating_system%29
+                return $"10.{version.Major - 4}";
+            }
+        }
+
+        public static Platform GetOSPlatform()
+        {
+            return _platform.Value;
+        }
+
+        private static string GetDistroId()
+        {
+            return _distroInfo.Value?.Id;
+        }
+
+        private static string GetDistroVersionId()
+        {
+            return _distroInfo.Value?.VersionId;
+        }
+
+        private static DistroInfo LoadDistroInfo()
+        {
+            // Sample os-release file:
+            //   NAME="Ubuntu"
+            //   VERSION = "14.04.3 LTS, Trusty Tahr"
+            //   ID = ubuntu
+            //   ID_LIKE = debian
+            //   PRETTY_NAME = "Ubuntu 14.04.3 LTS"
+            //   VERSION_ID = "14.04"
+            //   HOME_URL = "http://www.ubuntu.com/"
+            //   SUPPORT_URL = "http://help.ubuntu.com/"
+            //   BUG_REPORT_URL = "http://bugs.launchpad.net/ubuntu/"
+            // We use ID and VERSION_ID
+
+            if (File.Exists("/etc/os-release"))
+            {
+                var lines = File.ReadAllLines("/etc/os-release");
+                var result = new DistroInfo();
+                foreach (var line in lines)
+                {
+                    if (line.StartsWith("ID=", StringComparison.Ordinal))
+                    {
+                        result.Id = line.Substring(3).Trim('"', '\'');
+                    }
+                    else if (line.StartsWith("VERSION_ID=", StringComparison.Ordinal))
+                    {
+                        result.VersionId = line.Substring(11).Trim('"', '\'');
+                    }
+                }
+                return result;
+            }
+            return null;
+        }
+
+        // I could probably have just done one method signature and put the #if inside the body but the implementations
+        // are just completely different so I wanted to make that clear by putting the whole thing inside the #if.
+#if NET451
+        private static Platform DetermineOSPlatform()
+        {
+            var platform = (int)Environment.OSVersion.Platform;
+            var isWindows = (platform != 4) && (platform != 6) && (platform != 128);
```

Comment:
This should RuntimeEnvironmentHelper.IsWindows to avoid duplicating the same code